### PR TITLE
Add `brew sponsors` command

### PR DESCRIPTION
**Pull request by jonchang**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/brew/blob/HEAD/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/brew/pulls) for the same change?
- [x] Have you added an explanation of what your changes do and why you'd like us to include them?
- [x] Have you written new tests for your changes? [Here's an example](https://github.com/Homebrew/brew/blob/HEAD/Library/Homebrew/test/PATH_spec.rb).
- [x] Have you successfully run `brew style` with your changes locally?
- [x] Have you successfully run `brew tests` with your changes locally?

-----

Adds a new hidden developer command, `brew sponsors`, that queries the GitHub API to get a list of our supporters on GitHub Sponsors. 

Ideally this would be automatically updated via continuous integration, but this API requires a personal access token with scopes `admin:org` and `user`, which are some pretty big scopes and makes me a bit worried about security. Instead, someone with Owner permissions on the Homebrew org should run this every so often and update the README with its output. Suggestions for automation are welcome.

### CODE REVIEW

File: Library/Homebrew/dev-cmd/sponsors.rb [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+module Homebrew
+  module_function
+
+  def sponsors_args
+    Homebrew::CLI::Parser.new do
+      usage_banner <<~EOS
+        `sponsors`
+
+        Print a summary of Homebrew's GitHub Sponsors, suitable for pasting into a README.
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/sponsors.rb:14]:
```suggestion
        Print a markdown summary of Homebrew's GitHub Sponsors, suitable for pasting into a README.
```

File: Library/Homebrew/dev-cmd/sponsors.rb [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+      EOS
+      hide_from_man_page!
```
Comment (MikeMcQuaid) [inline Library/Homebrew/dev-cmd/sponsors.rb:16]: I'd just leave this in the manpage, tbh. No harm in having it there.

**Comment by MikeMcQuaid** [comment]
> Instead, someone with Owner permissions on the Homebrew org should run this every so often and update the README with its output.

Maybe someone on the PLC could have this be a recurring reminder?

**Comment by jonchang** [comment]
Yeah, on the first of each month or thereabouts would be ideal.

**Comment by jonchang** [comment]
Added a test to please codecov.

**Comment by jonchang** [comment]
Dunno what codecov wants but I'm just gonna merge this. Thanks for the review!

**Pull request merged**